### ISSUE TITLE
Stacks block commitments incorrectly assign change outputs as pox transfers 

### ISSUE THREAD

**Issue by aulneau** [hirosystems/chainhook#496]
**Describe the bug**

For the block commitments predicate, burn related pox blocks are incorrectly assigning change outputs as pox transfer events:

```json
       "stacks_operations": [
          {
            "block_committed": {
              "block_hash": "0x5b7e468df57b9d0208b67ffd3bc145a48e200270afb94fc8b11d910760276a01",
              "mining_address_post_commit": null,
              "mining_sats_left": 0,
              "pox_cycle_index": 74,
              "pox_cycle_length": 2200,
              "pox_cycle_position": 997,
              "pox_sats_burnt": 501000,
              "pox_sats_transferred": [
                {
                  "amount": 626173985,
                  "recipient_address": "1UiW61F9xnBnnC1J69aPJhFm3ZXvZEkCm"
                }
              ]
            }
          }
        ]
```

Here is the entire output for a block commitment event for [block 829847](https://mempool.space/block/000000000000000000025be4afc758baff0b5199dec236cfe444f1fda607ce67), and the related transaction: https://mempool.space/tx/151bc0d6ced51f3f7675f0caa60769543f2ec5962f0fe271c14b8c5feca466cf

https://gist.github.com/aulneau/f612837d61cbcc8e12d6d0a4c9ef8c63

seems possible this condition is incorrect: https://github.com/hirosystems/chainhook/blob/489b01458c35b9ed7db47b7c1c522e637367f7be/components/chainhook-sdk/src/indexer/bitcoin/mod.rs#L514

or maybe this logic?

https://github.com/hirosystems/chainhook/blob/489b01458c35b9ed7db47b7c1c522e637367f7be/components/chainhook-sdk/src/indexer/bitcoin/mod.rs#L551-L557

**Comment by lgalabru** [comment]
Thanks @aulneau, we're looking into this.

**Comment by MicaiahReid** [comment]
Hey @aulneau, great find on this bug!

I have a fix out in PR #499. Would you be able to checkout the branch `fix-block-commitment`, rebuild Chainhook, and confirm that things are working as expected?

**Comment by aulneau** [comment]
> Hey @aulneau, great find on this bug!
> 
> I have a fix out in PR #499. Would you be able to checkout the branch `fix-block-commitment`, rebuild Chainhook, and confirm that things are working as expected?

Unfortunately I do not have a bitcoin node set up -- I'm using the hiro platform for any bitcoin specific predicates
